experiment/gerrit-onboarder/onboarder.go

```diff
@@ -341,73 +349,131 @@ func ensureProjectConfig(config, host, cur_branch, groupName string) (string, er
+	if err := execInDir(os.Stdout, os.Stderr, workDir, "git", "switch", "-c", branch); err != nil {
 		return fmt.Errorf("failed to switch to new branch, %w", err)
 	}
 
 	return nil
 }
 
+func execInDir(stdout, stderr io.Writer, dir string, cmd string, args ...string) error {
+	(&logrus.Logger{
+		Out:       os.Stderr,
+		Formatter: logrus.StandardLogger().Formatter,
+		Hooks:     logrus.StandardLogger().Hooks,
+		Level:     logrus.StandardLogger().Level,
+	}).WithField("dir", dir).
+		WithField("cmd", cmd).
+		// The default formatting uses a space as separator, which is hard to read if an arg contains a space
+		WithField("args", fmt.Sprintf("['%s']", strings.Join(args, "', '"))).
+		Info("running command")
+
+	c := exec.Command(cmd, args...)
+	c.Dir = dir
+	c.Stdout = stdout
+	c.Stderr = stderr
+	return c.Run()
+}
+
+func createCR(workDir string) error {
+	diff, err := getDiff(workDir)
+	if err != nil {
+		return err
+	}
+	if diff == "" {
+		logrus.Info("No changes made. Returning without creating CR")
+		return nil
+	}
+	commitMessage := fmt.Sprintf("Grant the Prow cluster read and label permissions\n\nChange-Id: I%s", bumper.GitHash(fmt.Sprintf("%d", rand.Int())))
+	if err := execInDir(os.Stdout, os.Stderr, workDir, "git", "commit", "-a", "-v", "-m", commitMessage); err != nil {
+		return fmt.Errorf("unable to commit: %w", err)
+	}
+	if err := execInDir(os.Stdout, os.Stderr, workDir, "git", "push", "origin", "HEAD:refs/for/refs/meta/config"); err != nil {
+		return fmt.Errorf("unable to push: %w", err)
+	}
+	return nil
+}
+
+func getDiff(workDir string) (string, error) {
+	var diffBuf bytes.Buffer
+	var errBuf bytes.Buffer
+	if err := execInDir(&diffBuf, &errBuf, workDir, "git", "diff"); err != nil {
+		return "", fmt.Errorf("diffing previous bump: %v -- %s", err, errBuf.String())
+	}
+	return diffBuf.String(), nil
+}
+
 func main() {
 	o, err := parseAndValidateOptions()
 	if err != nil {
 		logrus.Fatal(err)
 	}
 
+	var workDir string
+	if o.local {
+		workDir, err = os.Getwd()
+		if err != nil {
+			logrus.Fatal(err)
+		}
+	} else {
+		workDir, err := ioutil.TempDir("", "gerrit_onboarding")
+		if err != nil {
+			logrus.Fatal(err)
+		}
+		defer os.RemoveAll(workDir)
+
+		if err := execInDir(os.Stdout, os.Stderr, workDir, "git", "clone", fmt.Sprintf("sso://%s/%s", o.host, o.repo)); err != nil {
+			logrus.Fatal(fmt.Errorf("failed to clone sso://%s/%s %w", o.host, o.repo, err))
+		}
+
+		workDir = path.Join(workDir, o.repo)
+	}
+
 	rand.Seed(time.Now().UTC().UnixNano())
 	branchName := fmt.Sprintf("gerritOnboarding_%d", rand.Int())
 
-	if err = fetchMetaConfig(o.host, o.repo, branchName); err != nil {
+	if err = fetchMetaConfig(o.host, o.repo, branchName, workDir); err != nil {
 		logrus.Fatal(err)
 	}
 
-	groupsChanged, err := updateGroups(o.uuID, o.groupName)
-	if err != nil {
+	if err = updatePojectConfig(workDir, o.host, branchName, o.groupName); err != nil {
```

Comment:
branch is already checked out in `fetchMetaConfig`, so I'm not super convinced that it's required here

Reply:
In another word, probably should just make branchname a local var in `fetchMetaConfig`

Reply:
I believe it is required we pass the branch name to updateProjectConfig for use in VerifyInTree. We keep track of the current branch so that we can create new branches when moving down the inheritance tree and be able to return to the original branch.

Reply:
I thought we are cloning in different dirs now, so might not need to switch branch again?